kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/GwtCertificatesServiceImpl.java

```diff
@@ -163,7 +163,7 @@ public List<GwtKeystoreEntry> listEntries() throws GwtKuraException {
                         Certificate[] chain = pke.getCertificateChain();
 
                         if (chain.length > 0) {
-                            Certificate leaf = chain[chain.length - 1];
+                            Certificate leaf = chain[0];
```

Comment:
Just a doubt here: I'm not sure if there might exist a chain with a CA expiring before the Private Key.
If that's the case, instead of choosing the first certificate of the chain, we should loop on the entire chain and show the first expiration date.

Reply:
I believe the most intersting date to see is the leaf, so I would use that date